### ISSUE TITLE
waypoint export with pictures

### ISSUE THREAD

**Issue by lada04** [bfabiszewski/ulogger-server#235]
Hi, 
I really enjoy ulogger-server together with ulogger-android, many thanks for the development and maintenance!

WIth the android app it is possible to create waypoints and assign pictures that are uploaded to the server and displayed nicely. 
However, these pictures are not exported. While gpx doesn't seem to support such a feature, would it be possible to include it in the kml export, e.g. as photo overlays? 

Thank you, 
Hartmut

**Comment by Nadyita** [comment]
It's possible to reference local images with the `link` element in waypoints. So something like this:
```xml
<gpx …>
 <wpt lat="52.00000000" lon="9.00000000">
   <ele>100.0</ele>
   <time>2024-09-24T11:25:07Z</time>
   <name>Test image</name>
   <link href="4_66f2a18fce628.jpg">
     <name>4_66f2a18fce628.jpg</name>
     <type>image/jpeg</type>
   </link>
 </wpt>
  <trk>
    …
  </trk>
</gpx>
```

I second this request, I'd love to be able to download the GPX track including images as a ZIP file :smiley:

**Comment by Nadyita** [comment]
Actually, the GPX export only exports the track, it doesn't export the waypoints at all. While at it, it would be nice to have these waypoints as well. Currently, the waypoints are only set as `<desc>` in the `<trkpt>`-element, but this doesn't show anywhere.

**Comment by Nadyita** [comment]
@bfabiszewski I actually implemented this as a proof of concept and the corresponding ZIP-file works nicely. The problem is that it would require the zip extension to work. Would you like me to open a PR, or is this project asleep for now?

**Comment by bfabiszewski** [comment]
> Actually, the GPX export only exports the track, it doesn't export the waypoints at all. While at it, it would be nice to have these waypoints as well. Currently, the waypoints are only set as `<desc>` in the `<trkpt>`-element, but this doesn't show anywhere.

Yes, currently the client app does not recognize between track points and waypoints. A waypoint is just a manually added trackpoint. A `desc` element in `trkpt` is valid. What do you mean by "doesn't show anywhere"?
Anyway we could export track points with comments or images as waypoints.

**Comment by bfabiszewski** [comment]
> @bfabiszewski I actually implemented this as a proof of concept and the corresponding ZIP-file works nicely. The problem is that it would require the zip extension to work. Would you like me to open a PR, or is this project asleep for now?

This would be nice to have. But this should be optional. So that users can chose their option. 
For example I only view images in ulogger-server. I occasionally export gpx to quickly view it in another application, eg. osmand. But I just want a lightweight gpx file without images. I am also not sure which apps would support such gpx file with locally referenced images.
On the other hand this could be useful for moving files with images from one ulogger server to another one. Now I can export the track on one server and import on another, but only without images. This would require to also add support for importing such zip files.

There is still one important issue. I started rewriting the whole server. If you want to make a PR for this feature it would be nice to have it already on the new version. I can publish what I have by now into a separate branch, but it is not functional yet. I mostly refactored backend. I still need some time to adjust the frontend and put it together. 
So the question is if you want to merge your feature into this beta branch. 
Anyway I will try to cleanup the code a bit and upload the beta branch on the weekend.

**Comment by Nadyita** [comment]
I don't think it makes sense for me to rewrite it to work with your beta code. For now, it works for me, and if @lada04 needs it as well, I can give them the modified `export.php`, but I'd rather wait for you to have something testable ready which I will then use as a basis. If your branch has a stable enough API, I'll gladly start contributing to it :smile_cat: 

> This would be nice to have. But this should be optional. So that users can chose their option.

Yes, I'd implement it as a third option "gpx + pics" or something like that.

**Comment by lada04** [comment]
Thanks, Nadyita, I'd be glad to test the modified `export.php` on my server with the current version!
